src/react/components/common/tagInput/tagInput.tsx

```diff
@@ -614,16 +634,71 @@ export class TagInput extends React.Component<ITagInputProps, ITagInputState> {
     private getTypeSubMenuItems = (): IContextualMenuItem[] => {
         const tag = this.state.selectedTag;
         const types = Object.values(FieldType);
-
-        return types.map((type) => {
-            return {
-                key: type,
-                text: type,
-                canCheck: true,
-                isChecked: type === tag.type,
-                onClick: this.onTypeSelect,
-            } as IContextualMenuItem;
-        });
+        if (tag && tag.documentCount === 0) {
+            // Free tag - user can assign any type
+            if (tag.format === FieldFormat.NotSpecified) {
+                return types.map((type) => {
+                    return {
+                        key: type,
+                        text: type,
+                        canCheck: true,
+                        isChecked: type === tag.type,
+                        onClick: this.onTypeSelect,
+                    } as IContextualMenuItem;
+                });
+            } else {
```

Comment:
I'm confused, these if and else blocks return the same thing?

Reply:
I does. Kun said we want do something with empty but not free tag. Thats is why I put temp comment below.